system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp

```diff
@@ -29,29 +29,57 @@ RoutingAdaptor::RoutingAdaptor() : Node("routing_adaptor")
   const auto adaptor = component_interface_utils::NodeAdaptor(this);
   adaptor.init_cli(cli_route_);
   adaptor.init_cli(cli_clear_);
-  route_points_ = std::make_shared<SetRoutePoints::Service::Request>();
+  adaptor.init_sub(
+    sub_state_, [this](const RouteState::Message::ConstSharedPtr msg) { state_ = msg->state; });
+
+  const auto rate = rclcpp::Rate(5.0);
+  timer_ = rclcpp::create_timer(
+    this, get_clock(), rate.period(), std::bind(&RoutingAdaptor::on_timer, this));
+
+  state_ = RouteState::Message::UNKNOWN;
+  route_ = std::make_shared<SetRoutePoints::Service::Request>();
```

Comment:
Setting nullptr first sounds better to me.
If you don't think it's good, just ignore please.

Reply:
This value is always used and does not need to be nullptr.

Reply:
Then why do you make route_ shared pointer?

Reply:
Because the `async_send_request` function only supports shared pointer.

Reply:
Got it. Thank you